File: kotlin/src/test/kotlin/PrimeTableTest.kt
import kotlin.test.assertEquals
import kotlin.test.assertTrue
import org.junit.Test

import cpalgo.primes.*

val largePrimes = primes(2000000)

class PrimeTableTest {
  @Test
  fun testPrimesLarge() {
    assertEquals(largePrimes.takeWhile { it < 1_000_000 }.map { it.toLong() }.sum(), 37550402023L)
    assertEquals(largePrimes.map { it.toLong() }.sum(), 142913828922L)
  }
  @Test
  fun testPrimesSmall() {
    for (i in 1 .. 100) {
      val l1 = largePrimes.takeWhile { it < i }.toIntArray ()
      val l2 = primes(i)
      assertTrue (l1.contentEquals(l2))
    }
  }
}

File: scala/src/test/scala/PrimeTableTest.scala
package com.github.antma.cpalgo
import org.specs2.mutable.Specification

class PrimeTableTest extends Specification {
  "PrimeTable" should {
    "sum of primes <1000000" in {
      new PrimeTable(1000000).primes.foldRight(0L) { _ + _ } must_== 37550402023L
    }
    "primes below 200 tests" in {
      val p = List(2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89,
        97, 101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193,
        197, 199)
      (1 until 200).forall { n =>
        new PrimeTable(n).primes.sameElements(p.takeWhile { _ < n })
      } must beTrue
    }
  }
}
